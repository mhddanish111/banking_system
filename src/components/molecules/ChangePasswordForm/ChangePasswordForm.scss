@import "../../../styles/colors.scss";
@import "../../../styles/boarder.scss";
@import "../../../styles/spacing.scss";
@import "../../../styles/mixins.scss";
.change-password-form {
  display: flex;
  flex-direction: column;
  background-color: $background-color_2;
  border: $border-medium;
  border-radius: $border-radius;
  box-shadow: $box-shadow-focus;
  padding: $space-l;
  width: 100%;
  justify-content: center;
  @include desktop{
      width: 50%;
  }
  @include tablet {
      width: 80%;
  }
  @include mobile {
    padding: $space-s;
  }
  .form-action {
    display: flex;
    justify-content: center;
    margin-top: $space-xs;
  }
  .error-message {
    color: $color-red-50;
    margin-bottom: $space-xxs;
  }
  .error-block {
    width: 100%;
    margin-bottom: $space-xxs;
    .error-message-container {
      margin: 0;
    }
  }
}
