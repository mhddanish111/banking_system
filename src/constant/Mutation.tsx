import { gql } from '@apollo/client';

export const CREATE_USER = gql`
  mutation CreateUser($userInput: UserInput) {
    createUser(userInput: $userInput) {
      firstName
      lastName
      message
    }
  }
`;

export const CHANGE_PASSWORD = gql`
  mutation ChangePassword($changePasswordInput: ChangePasswordInput) {
    changePassword(changePasswordInput: $changePasswordInput) {
      message
      success
      data {
        firstName
      }
    }
  }
`;

export const OPEN_ACCOUNT = gql`
  mutation CreateAccount($accountInput: AccountInput) {
    createAccount(accountInput: $accountInput) {
      _id
      accountNumber
      amount
      accountType
      categoryName
      limit
      minAmount
      maxAmount
      accountTypeId
      categoryId
      userId
      message
      type
      debitCardNumber
      debitCardCSVNumber
      expiryYear
      expiryMonth
    }
  }
`;

export const DEPOSIT = gql`
  mutation Deposit($depositInput: DepositInput) {
    deposit(depositInput: $depositInput) {
      transactionResult {
        _id
      }
      accounts {
        _id
        type
        accountType
        categoryName
        accountNumber
        amount
        minAmount
        maxAmount
        limit
        charge
        debitCardNumber
        debitCardCSVNumber
        expiryYear
        expiryMonth
        users {
          _id
          firstName
          lastName
        }
      }
    }
  }
`;

export const WITHDRAWAL = gql`
  mutation Withdrawal($withdrawalInput: WithdrawalInput) {
    withdrawal(withdrawalInput: $withdrawalInput) {
      transactionResult {
        _id
      }
      accounts {
        _id
        type
        accountType
        categoryName
        accountNumber
        amount
        minAmount
        maxAmount
        limit
        charge
        debitCardNumber
        debitCardCSVNumber
        expiryYear
        expiryMonth
        users {
          _id
          firstName
          lastName
        }
      }
    }
  }
`;
