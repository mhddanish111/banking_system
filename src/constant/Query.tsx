import { gql } from '@apollo/client';

export const LOGIN_USER = gql`
  query Login($logInInput: LogInInput) {
    login(logInInput: $logInInput) {
      firstName
      lastName
      userId
      token
      isValid
      message
      userType
    }
  }
`;

export const USER_DETAILS = gql`
  query UserDetails {
    userDetails {
      _id
      firstName
      lastName
      email
      mobile
      occupation
      income
      aadhaarNumber
      panNumber
      govtId
      govtIdNumber
      address
      city
      state
      postalCode
      isActive
      message
    }
  }
`;

export const GET_ACCOUNT_TYPES = gql`
  query AccountTypes {
    accountType {
      _id
      name
      type
      isActive
      category {
        _id
        name
        benefit
        isActive
        minAmount
        maxAmount
        limit
        charge
      }
    }
  }
`;

export const GET_ACCOUNTS = gql`
  query Accounts {
    accounts {
      _id
      type
      accountType
      categoryName
      accountNumber
      amount
      minAmount
      maxAmount
      limit
      charge
      debitCardNumber
      debitCardCSVNumber
      expiryYear
      expiryMonth
      users {
        _id
        firstName
        lastName
      }
    }
  }
`;

export const GET_ACCOUNT_WITH_TYPE = gql`
  query AccountWithType($accountWithTypeInput: AccountWithTypeInput) {
    accountWithType(accountWithTypeInput: $accountWithTypeInput) {
      _id
      type
      accountType
      categoryName
      accountNumber
      amount
      minAmount
      maxAmount
      limit
      charge
      debitCardNumber
      debitCardCSVNumber
      expiryYear
      expiryMonth
      users {
        _id
        firstName
        lastName
        aadhaarNumber
        panNumber
      }
    }
  }
`;

export const GET_TRANSACTION = gql`
  query Transaction($transactionInput: TransactionInput) {
    transaction(transactionInput: $transactionInput) {
      count {
        offset
        limit
        totalCount
      }
      result {
        _id
        transactionId
        accountNumber
        amount
        withdrawalAmount
        depositAmount
        transactionDate
        accountType
        categoryName
        transactionType
      }
    }
  }
`;
