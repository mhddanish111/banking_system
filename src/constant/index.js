export const USER_KEY = 'user_key';

export const AccountTypeList = [
  { label: 'Accounts', value: 'Accounts' },
  { label: 'Cards', value: 'Cards' },
];
