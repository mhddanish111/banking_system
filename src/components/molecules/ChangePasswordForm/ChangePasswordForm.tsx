import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import { useMutation } from '@apollo/client';
import Button from '../../atoms/Button';
import ErrorMessage from '../../atoms/ErrorMessage';
import { CHANGE_PASSWORD } from '../../../constant/Mutation';
import { FormData } from '../../../type/UserType';
import './ChangePasswordForm.scss';

const ChangePasswordForm: React.FC<any> = () => {
  const [submitChangePassword] = useMutation(CHANGE_PASSWORD);
  const [newPassword, setNewPassword] = useState<String>('');
  const [errorMessage, setErrorMessage] = useState<String>('');
  const [isSuccess, setIsSuccess] = useState<Boolean>(false);
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<FormData>({ mode: 'onChange' });
  const getErrorMessage = (message: String | undefined) => {
    return message && <p className="error-message">{message}</p>;
  };
  const onSubmit = (data: FormData, e: any) => {
    e.preventDefault();
    const { oldPassword, newPassword, confirmPassword } = data;
    if (oldPassword === newPassword) {
      return setErrorMessage('Old and New password should not be same');
    }
    if (oldPassword === newPassword || newPassword !== confirmPassword) {
      return setErrorMessage('Password does not match');
    }
    const changePasswordData = {
      oldPassword,
      newPassword,
    };
    submitChangePassword({
      variables: { changePasswordInput: changePasswordData },
    })
      .then((res: any) => {
        const { message, success } = res?.data?.changePassword;
        if (!success) {
          return setErrorMessage(message);
        }
        return setIsSuccess(success);
      })
      .catch((err) => {
        return setErrorMessage(err?.message);
      });
  };
  const closeError = (e: any) => {
    e.preventDefault();
    setErrorMessage('');
    setIsSuccess(false);
  };
  return (
    <form
      name="change-password-form"
      className="container change-password-form"
      aria-label="change password form"
      onSubmit={handleSubmit(onSubmit)}
    >
      <div className="error-block">
        {errorMessage && (
          <ErrorMessage
            message={errorMessage}
            className="alert"
            onClick={closeError}
          />
        )}
        {isSuccess && (
          <ErrorMessage
            message="Your Password have been changed successfully."
            className="success"
            onClick={closeError}
          />
        )}
      </div>
      <div className="row change-password-container">
        <div className="col-xs-12">
          <label htmlFor="oldPassword" aria-label="old password">
            Old Password
          </label>
          <input
            type="password"
            id="oldPassword"
            autoComplete="off"
            aria-label="old password input"
            {...register('oldPassword', {
              required: 'Old Password is required',
              maxLength: {
                value: 15,
                message: 'You exceeded max length of 15 !',
              },
            })}
          />
          {getErrorMessage(errors?.oldPassword?.message)}
        </div>

        <div className="col-xs-12">
          <label htmlFor="newPassword" aria-label="new password">
            New Password
          </label>
          <input
            type="password"
            id="newPassword"
            autoComplete="off"
            aria-label="new password input"
            {...register('newPassword', {
              onBlur: (e) => setNewPassword(e.target.value),
              required: 'New Password is required',
              maxLength: {
                value: 15,
                message: 'You exceeded max length of 15 !',
              },
            })}
          />
          {getErrorMessage(errors?.newPassword?.message)}
        </div>

        <div className="col-xs-12">
          <label htmlFor="confirmPassword" aria-label="confirm password">
            Confirm Password
          </label>
          <input
            type="password"
            id="confirmPassword"
            autoComplete="off"
            aria-label="confirm password input"
            {...register('confirmPassword', {
              required: 'Confirm Password is required',
              validate: (value) =>
                value === newPassword || 'Password does not match',
              maxLength: {
                value: 15,
                message: 'You exceeded max length of 15 !',
              },
            })}
          />
          {getErrorMessage(errors?.confirmPassword?.message)}
        </div>
        <div className="col-xs-12 form-action">
          <Button type="submit">Sign Up</Button>
        </div>
      </div>
    </form>
  );
};

export default ChangePasswordForm;
