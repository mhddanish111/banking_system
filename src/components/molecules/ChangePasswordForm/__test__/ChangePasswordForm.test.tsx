import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';
import ChangePasswordForm from '../ChangePasswordForm';
import { CHANGE_PASSWORD } from '../../../../constant/Mutation';

describe('<ChangePasswordForm />', () => {
  const changePasswordData = {
    oldPassword: 'test',
    newPassword: 'test@123',
  };
  const mocks = [
    {
      request: {
        query: CHANGE_PASSWORD,
        variables: {
          changePasswordInput: changePasswordData,
        },
      },
      result: {
        data: {
          changePassword: {
            message: 'test',
            success: true,
          },
        },
      },
    },
  ];
  const wrapper = renderer.create(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ChangePasswordForm />
    </MockedProvider>,
  ).toJSON;
  it('should render the ChangePasswordForm Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
