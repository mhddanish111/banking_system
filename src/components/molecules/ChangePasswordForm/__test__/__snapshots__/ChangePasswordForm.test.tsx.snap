// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePasswordForm /> should render the ChangePasswordForm Component correctly 1`] = `[Function]`;
